lib/tests/streamlit/state/session_state_test.py

```diff
@@ -233,6 +233,23 @@ def test_widget_presence(self):
         assert "foo" in state
         assert state.foo == "foo"
 
+    def test_widget_outputs_dont_alias(self):
+        color = st.select_slider(
+            "Select a color of the rainbow",
+            options=[
+                ["red", "orange"],
+                ["yellow", "green"],
+                ["blue", "indigo"],
+                ["violet"],
+            ],
+            key="color",
+        )
+
+        color.append("purple")
+
+        ctx = get_report_ctx()
+        assert ctx.session_state._initial_widget_values["color"] is not color
```

Comment:
For this assertion, maybe im missing something but is checking that they're not the same a good test?
I am thinking that checking that we should check that purple got added?

Reply:
The purpose of this assertion is to verify that `ctx.session_state._initial_widget_values["color"]` does _not_ get purple added. The test would perhaps be simpler if I left off the purple part, because I used `is` rather than `==` so that it would check reference identity rather than value equality, so the purple append is unnecessary